### PR TITLE
🚨🚨🚨 fix(Mask2Former): torch export 🚨🚨🚨

### LINKED ISSUE #34390: [mask2former] torch.export error for Mask2Former
### System Info

- `transformers` version: 4.46.0.dev0
- Platform: Linux-6.8.0-47-generic-x86_64-with-glibc2.35
- Python version: 3.11.9
- Huggingface_hub version: 0.25.2
- Safetensors version: 0.4.5
- Accelerate version: not installed
- Accelerate config: not found
- PyTorch version (GPU?): 2.4.1+cu121 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using distributed or parallel set-up in script?: <fill in>
- Using GPU in script?: <fill in>
- GPU type: NVIDIA GeForce RTX 4090

### Who can help?

@amyeroberts, @qubvel, @ylacombe

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

```python
import torch
from transformers import Mask2FormerForUniversalSegmentation

model = Mask2FormerForUniversalSegmentation.from_pretrained(
    "facebook/mask2former-swin-base-coco-panoptic", torchscript=True
)

scripted_model = torch.export.export(model, args=(torch.randn(1, 3, 800, 1280),))
```
which causes
```
UserError: Could not extract specialized integer from data-dependent expression u0 (unhinted: u0).  (Size-like symbols: none)

Potential framework code culprit (scroll up for full backtrace):
  File "/home/philkuz/.pyenv/versions/3.11.9/envs/gml311/lib/python3.11/site-packages/torch/_dynamo/utils.py", line 2132, in run_node
    return node.target(*args, **kwargs)

For more information, run with TORCH_LOGS="dynamic"
For extended logs when we create symbols, also add TORCHDYNAMO_EXTENDED_DEBUG_CREATE_SYMBOL="u0"
If you suspect the guard was triggered from C++, add TORCHDYNAMO_EXTENDED_DEBUG_CPP=1
For more debugging help, see https://docs.google.com/document/d/1HSuTTVvYH1pTew89Rtpeu84Ht3nQEFTYhAX3Ypa_xJs/edit?usp=sharing

User Stack (most recent call last):
  (snipped, see stack below for prefix)
  File "/home/philkuz/dev/transformers/src/transformers/models/mask2former/modeling_mask2former.py", line 2499, in forward
    outputs = self.model(
  File "/home/philkuz/.pyenv/versions/3.11.9/envs/gml311/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/philkuz/dev/transformers/src/transformers/models/mask2former/modeling_mask2former.py", line 2270, in forward
    pixel_level_module_output = self.pixel_level_module(
  File "/home/philkuz/.pyenv/versions/3.11.9/envs/gml311/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/philkuz/dev/transformers/src/transformers/models/mask2former/modeling_mask2former.py", line 1395, in forward
    decoder_output = self.decoder(backbone_features, output_hidden_states=output_hidden_states)
  File "/home/philkuz/.pyenv/versions/3.11.9/envs/gml311/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/philkuz/dev/transformers/src/transformers/models/mask2former/modeling_mask2former.py", line 1319, in forward
    encoder_outputs = self.encoder(
  File "/home/philkuz/.pyenv/versions/3.11.9/envs/gml311/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/philkuz/dev/transformers/src/transformers/models/mask2former/modeling_mask2former.py", line 1165, in forward
    reference_points = self.get_reference_points(spatial_shapes, valid_ratios, device=inputs_embeds.device)
  File "/home/philkuz/dev/transformers/src/transformers/models/mask2former/modeling_mask2former.py", line 1106, in get_reference_points
    torch.linspace(0.5, height - 0.5, height, dtype=valid_ratios.dtype, device=device),

For C++ stack trace, run with TORCHDYNAMO_EXTENDED_DEBUG_CPP=1
For more information about this error, see: https://pytorch.org/docs/main/generated/exportdb/index.html#constrain-as-size-example

from user code:
   File "/home/philkuz/dev/transformers/src/transformers/models/mask2former/modeling_mask2former.py", line 2499, in forward
    outputs = self.model(
  File "/home/philkuz/.pyenv/versions/3.11.9/envs/gml311/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/philkuz/dev/transformers/src/transformers/models/mask2former/modeling_mask2former.py", line 2270, in forward
    pixel_level_module_output = self.pixel_level_module(
  File "/home/philkuz/.pyenv/versions/3.11.9/envs/gml311/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/philkuz/dev/transformers/src/transformers/models/mask2former/modeling_mask2former.py", line 1395, in forward
    decoder_output = self.decoder(backbone_features, output_hidden_states=output_hidden_states)
  File "/home/philkuz/.pyenv/versions/3.11.9/envs/gml311/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/philkuz/dev/transformers/src/transformers/models/mask2former/modeling_mask2former.py", line 1319, in forward
    encoder_outputs = self.encoder(
  File "/home/philkuz/.pyenv/versions/3.11.9/envs/gml311/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/philkuz/dev/transformers/src/transformers/models/mask2former/modeling_mask2former.py", line 1165, in forward
    reference_points = self.get_reference_points(spatial_shapes, valid_ratios, device=inputs_embeds.device)
  File "/home/philkuz/dev/transformers/src/transformers/models/mask2former/modeling_mask2former.py", line 1106, in get_reference_points
    torch.linspace(0.5, height - 0.5, height, dtype=valid_ratios.dtype, device=device),
 ```

### Expected behavior

torch.export works for this model.

### PR DESCRIPTION
**Pull request by philkuz**
# What does this PR do?

Fixes #34390 (issue)

Mask2Former modeling had a set of issues that prevented `torch.export` from working. This PR addresses ~3 individual problems that prevented the model from working. I took direction from a few PRs
that made similar changes:
1. [fix(Wav2Vec2ForCTC): torch export #34023](https://github.com/huggingface/transformers/pull/34023) for the attention mask modification to prevent a RuntimeError during torch.export
2. [Optim deformable detr #33600](https://github.com/huggingface/transformers/pull/33600) for passing around [spatial_shapes as a list](https://github.com/huggingface/transformers/pull/33600/files#diff-137bda311c71a6ad1765c311a04366dc1b547213c95795752ddb11e39466bbabR701) and as a tensor [and for the if statement fix](https://github.com/huggingface/transformers/pull/33600/files#diff-137bda311c71a6ad1765c311a04366dc1b547213c95795752ddb11e39466bbabR711)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?
@amyeroberts, @qubvel, @ylacombe (tagging because you looked at #34023)

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

 ## Testing:
 Run this once in `main` and then once with this branch. Ensure the testing.all_close() works.
 ```python
 import os

import numpy as np
import requests
import torch
from PIL import Image
from transformers import AutoImageProcessor, Mask2FormerForUniversalSegmentation


def get_test_image():
    url = "http://images.cocodataset.org/val2017/000000039769.jpg"
    image = Image.open(requests.get(url, stream=True).raw)
    return image


processor = AutoImageProcessor.from_pretrained(
    "facebook/mask2former-swin-base-coco-panoptic"
)
model = Mask2FormerForUniversalSegmentation.from_pretrained(
    "facebook/mask2former-swin-base-coco-panoptic"
)


results = model(**processor(images=get_test_image(), return_tensors="pt"))

results = processor.post_process_panoptic_segmentation(results)

results_file = "segmentation_results.pt"
if os.path.exists(results_file):
    old_segmentation = torch.load(results_file)
    np.testing.assert_array_almost_equal(results[0]["segmentation"], old_segmentation)
else:
    torch.save(results[0]["segmentation"], results_file)

scripted_model = torch.export.export(model, args=(torch.randn(1, 3, 800, 1280),))

```

I also visually compared the output masks and they look the same

### CODE REVIEW

File: src/transformers/models/mask2former/modeling_mask2former.py
Comment (yonigozlan) [inline src/transformers/models/mask2former/modeling_mask2former.py:1894]: Nice!

Comment (yonigozlan) [inline src/transformers/models/mask2former/modeling_mask2former.py:1894]: Nice!

File: src/transformers/models/mask2former/modeling_mask2former.py
Comment (yonigozlan) [inline src/transformers/models/mask2former/modeling_mask2former.py:1326]: Is this change necessary for torch compile/export? Does it impact performance? Just curious as this is also present in other models where a spatial_shapes_list is added, but it did not seem to break the graph or cause any problems with torch compile

**Review by yonigozlan — COMMENTED** [review]
Looks good to me, thanks for working on this! I just have a question regarding one of your changes

File: src/transformers/models/mask2former/modeling_mask2former.py
Comment (philkuz) [reply src/transformers/models/mask2former/modeling_mask2former.py:1326]: I'll investigate some more, but when I undo this line change, I start getting this error during torch.export
```
from user code:
   File "/home/philkuz/dev/transformers/src/transformers/models/mask2former/modeling_mask2former.py", line 2512, in forward
    outputs = self.model(
  File "/home/philkuz/.pyenv/versions/gml311/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/philkuz/dev/transformers/src/transformers/models/mask2former/modeling_mask2former.py", line 2283, in forward
    pixel_level_module_output = self.pixel_level_module(
  File "/home/philkuz/.pyenv/versions/gml311/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/philkuz/dev/transformers/src/transformers/models/mask2former/modeling_mask2former.py", line 1406, in forward
    decoder_output = self.decoder(backbone_features, output_hidden_states=output_hidden_states)
  File "/home/philkuz/.pyenv/versions/gml311/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1747, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/philkuz/dev/transformers/src/transformers/models/mask2former/modeling_mask2former.py", line 1355, in forward
    outputs = [
  File "/home/philkuz/dev/transformers/src/transformers/models/mask2former/modeling_mask2former.py", line 1356, in <listcomp>
    x.transpose(1, 2).view(batch_size, -1, spatial_shapes_list[i][0], spatial_shapes_list[i][1])
 ```
 
 which is connected through this:
 1. `level_start_index` fills in the `split_sizes` in L1347-1352
 2. `split_sizes` is an input to `torch.split` which outputs `encoder_output` in L1345 
 3. `encoder_output` is iterated over in the list comprehension of L1357-1359 which is where we get the bottom of the stack trace.
 
 Let me find the other uses of `spatial_shapes_list` and see what's different.

Comment (philkuz) [reply src/transformers/models/mask2former/modeling_mask2former.py:1326]: Tangent: `level_start_index` isn't used at all in the attention module and I can't find any other uses than to inform `split_sizes`

Comment (philkuz) [reply src/transformers/models/mask2former/modeling_mask2former.py:1326]: Some updates:
I've found the following models use a `level_start_index`:
1. deformable_detr
2. oneformer
3. rt_detr
this is not a full list, but I think helped me get an idea of what's going on.

neither deformable_detr nor rt_detr have a similar `torch.split` call which is a way to break apart the `multi-level` result of the decoder
image

`oneformer` has a simliar setup, but I also failed to export the model, likely because the issues are very similar to what I faced with mask2former. Happy to help make `oneformer` compatible for export if you would like!

But my conclusion is that `level_start_index` does need to change to be compatible for our specific use-cases. We could also simply rename `level_start_index` in this case to better match that we are using it for split rather than for `level_start_index` parameter, especially since we don't actually use this value in the modeling code.

**Comment by philkuz** [comment]
Hey @yonigozlan if you have a chance, I would love another review of this PR after addressing your comments!
Thanks

File: src/transformers/models/mask2former/modeling_mask2former.py
Comment (yonigozlan) [reply src/transformers/models/mask2former/modeling_mask2former.py]: Thanks for diving deeply in this issue and for the detailed explanation! You're right it looks like `level_start_index` is actually not used in this model, apart for building `split_sizes`, so we definitely shouldn't let it block `torch.export` :). Looking at other models using `level_start_index`, it seems that it is used only for the custom kernel `MultiScaleDeformableAttentionFunction`, which is not present in mask2former (don't know why, but that can be the object of another PR).
I agree with what you've done, I think we can leave the tensor version of `level_start_index` even if it's not used now, and open a PR in the future to add the custom kernel `MultiScaleDeformableAttentionFunction` that will use it.
As you explained, only oneformer should have a similar problem with `level_start_index`, so feel free to open a PR for that too!

**Comment by yonigozlan** [comment]
Looks great to me thanks again for fixing this! I think we should add a short comment just above the definition of `level_start_index_list` explaining why we do this (iterating over a tensor breaks torch.compile/export).
Please rebase on main and then ask a core maintainer for a final review :).

**Comment by philkuz** [comment]
Thanks Yoni!

Pinging @amyeroberts and @qubvel for a core maintainer review!

File: src/transformers/models/mask2former/modeling_mask2former.py
Comment (qubvel) [inline src/transformers/models/mask2former/modeling_mask2former.py:1349]:
```suggestion
        for height, width in spatial_shapes_list[:-1]:
            level_start_index_list.append(level_start_index_list[-1] + height * width)
```

File: src/transformers/models/mask2former/modeling_mask2former.py
Comment (qubvel) [inline src/transformers/models/mask2former/modeling_mask2former.py:1132]: Should we pass both the list and the tensor? Can't we convert it inside or are there any performance issues in that case?
cc @yonigozlan

**Review by qubvel — COMMENTED** [review]
Thanks for working on this!

It would be also nice to have a test that export works, maybe smth similar to [this](https://github.com/huggingface/transformers/commit/5392f12e1614383270ae8df524415a1f6b555773).

File: src/transformers/models/mask2former/modeling_mask2former.py
Comment (philkuz) [reply src/transformers/models/mask2former/modeling_mask2former.py:1132]: we could probably convert the list to the tensor inside this forward call without affecting performance. 

the spatial_shapes tensor is directly passed into multiple `encoder_layer`s of this module so this feels like the lowest we can go without changing the performance. Although I wonder how bad the performance impact will be if we never create a tensor and instead read the spatial_shapes_list here: https://github.com/philkuz/transformers/blob/78f384808a70d346cd8f1e8d314411510edec11d/src/transformers/models/mask2former/modeling_mask2former.py#L962

Comment (philkuz) [reply src/transformers/models/mask2former/modeling_mask2former.py:1132]: Wrote a quick torch benchmark based on where this tensor is actually used
```
import time
import torch

# Create sample data - list of 1000 tuples with 3 elements each
# https://github.com/philkuz/transformers/blob/78f384808a70d346cd8f1e8d314411510edec11d/src/transformers/models/mask2former/modeling_mask2former.py#L1219
num_feature_levels = 3  # hardcoded in the modeling code

# https://huggingface.co/facebook/mask2former-swin-small-coco-instance/blob/main/config.json#L2118
# https://huggingface.co/facebook/mask2former-swin-large-coco-instance/blob/main/config.json#L2118
config_encoder_layers = (
    6  # Set in the config, seems to be set at 6 across small and large models
)
data = [(i, i + 1) for i in range(num_feature_levels)]

# Run benchmark
num_iterations = 1000
total_time = 0

for _ in range(num_iterations):
    start = time.perf_counter()
    tensor = torch.tensor(data)
    for i in range(config_encoder_layers):
        offset_normalizer_tensor = torch.stack([tensor[..., 1], tensor[..., 0]], -1)
    end = time.perf_counter()
    total_time += end - start

avg_time = total_time / num_iterations
print(
    f"Average time from spatial_shapes tensor {num_iterations} iterations: {avg_time:.6f} seconds"
)


num_iterations = 1000
total_time = 0
for _ in range(num_iterations):
    start = time.perf_counter()
    for i in range(config_encoder_layers):
        offset_normalizer_tensor = torch.tensor([[d[1], d[0]] for d in data])
    end = time.perf_counter()
    total_time += end - start

avg_time = total_time / num_iterations
print(
    f"Average time from original list of tuples {num_iterations} iterations: {avg_time:.6f} seconds"
)
```

results:
```
Average time from spatial_shapes tensor 1000 iterations: 0.000064 seconds
Average time from original list of tuples 1000 iterations: 0.000033 seconds
```

I think we should just drop the tensor entirely, given that we really are dealing with a 3x2 sized list, seems like tensors just add cognitive and performance overhead.

**Comment by philkuz** [comment]
> Thanks for working on this!
> 
> It would be also nice to have a test that export works, maybe smth similar to [this](https://github.com/huggingface/transformers/commit/5392f12e1614383270ae8df524415a1f6b555773).

Added a test lemme know what you think.

File: src/transformers/models/mask2former/modeling_mask2former.py
Comment (philkuz) [reply src/transformers/models/mask2former/modeling_mask2former.py]: I ended up dropping the spatial_shapes tensor that's passed around. I still have the original because `level_start_index` still exists, but I think we should just remove `level_start_index` in a separate PR.

Comment (philkuz) [reply src/transformers/models/mask2former/modeling_mask2former.py]: I address that change here: https://github.com/huggingface/transformers/pull/34498

File: tests/models/mask2former/test_modeling_mask2former.py
Comment (qubvel) [inline tests/models/mask2former/test_modeling_mask2former.py:488]: Can you please update 

```python
from transformers.pytorch_utils import is_torch_greater_or_equal_2_4
```
instead of doing comparison with packaging

File: tests/models/mask2former/test_modeling_mask2former.py
Comment (qubvel) [inline tests/models/mask2former/test_modeling_mask2former.py:486]: it should be slow test -  decorated with `@slow`

**Review by qubvel — COMMENTED** [review]
Thanks for a quick iteration! A few more comments.

Please, push empty commit with message`[run_slow] mask2former` at the end to trigger all slow tests run for mask2former model

File: src/transformers/models/mask2former/modeling_mask2former.py
Comment (qubvel) [reply src/transformers/models/mask2former/modeling_mask2former.py]: Ok, great! Thanks a lot for benchmarking it!

File: tests/models/mask2former/test_modeling_mask2former.py
Comment (philkuz) [reply tests/models/mask2former/test_modeling_mask2former.py:486]: yep the testcase (the class) is already marked as slow: https://github.com/philkuz/transformers/blob/d08855c2e7e49878f8f80af9c590778f77dbe651/tests/models/mask2former/test_modeling_mask2former.py#L364

Comment (philkuz) [reply tests/models/mask2former/test_modeling_mask2former.py:486]:
```
tests/models/mask2former/test_modeling_mask2former.py::Mask2FormerModelIntegrationTest::test_export SKIPPED (test is slow)
```

File: src/transformers/models/mask2former/modeling_mask2former.py
Comment (philkuz) [reply src/transformers/models/mask2former/modeling_mask2former.py]: no problem

**Comment by philkuz** [comment]
> Thanks for a quick iteration! A few more comments.
> 
> Please, push empty commit with message`[run_slow] mask2former` at the end to trigger all slow tests run for mask2former model

You got it!

**Review by qubvel — APPROVED** [review]
LGTM, thanks!

File: src/transformers/models/mask2former/modeling_mask2former.py
Comment (ArthurZucker) [inline src/transformers/models/mask2former/modeling_mask2former.py:929]: changing the name is a breaking change, we should probably have a deprecation cycle no?

Comment (ArthurZucker) [reply src/transformers/models/mask2former/modeling_mask2former.py:929]: if not, we can add 🔴 as I think the motivation is strong enough.

**Review by ArthurZucker — COMMENTED** [review]
I am not 100% we have to break not, do you think we can keep supporting the old behaviour / at least have a deprecation cycle?

File: src/transformers/models/mask2former/modeling_mask2former.py
Comment (qubvel) [reply src/transformers/models/mask2former/modeling_mask2former.py:929]: Hmm, I suppose it is an internal module of the model, not sure if it is intended to be used elsewhere, let me know if I'm wrong

Comment (philkuz) [reply src/transformers/models/mask2former/modeling_mask2former.py:929]: Isn't changing the logic, but keeping both `spatial_shapes` and `spatial_shapes_list` already a breaking change?

Sure there's some question of whether users can rely on internal Modules of transformers models, but also if a user doesn't pass a value for `spatial_shapes_list` this code will fail as [L939](https://github.com/huggingface/transformers/pull/34393/files#diff-1fee617cdf6a827988f42ee7a5cde812e08a4867257e03684df091e03a4141deR939) will try to iterate over a None object.

BTW seems like the changes in #33600 already violate this contract([see this line](https://github.com/huggingface/transformers/pull/33600/files#diff-137bda311c71a6ad1765c311a04366dc1b547213c95795752ddb11e39466bbabR711))? I followed that PR as a guide on what I should change here.

It seems like the proper way forward is to add a 🔴  here.

Comment (philkuz) [reply src/transformers/models/mask2former/modeling_mask2former.py:929]: What's the path forward @ArthurZucker ?

Comment (philkuz) [reply src/transformers/models/mask2former/modeling_mask2former.py:929]: Bumping this again after the weekend. I'm not familiar with Huggingface's policies of 
1. What qualifies as a breaking change?
2. What the release process is?

Could you provide a recommendation on where we should take this PR?

**Comment by qubvel** [comment]
soft ping @ArthurZucker, regarding your last comment, I'm not sure it can be done without breaking changes, but as soon as it is the internal module of the model I suppose it is fine to change its signature. 🚨 can be added to PR message if it is still required

**Comment by philkuz** [comment]
> soft ping @ArthurZucker, regarding your last comment, I'm not sure it can be done without breaking changes, but as soon as it is the internal module of the model I suppose it is fine to change its signature. 🚨 can be added to PR message if it is still required

Hi @ArthurZucker @qubvel, 

Hope you're doing well, could you find some time today or tomorrow to provide guidance on this PR? Would love to check it off of my list!

**Comment by philkuz** [comment]
> > soft ping @ArthurZucker, regarding your last comment, I'm not sure it can be done without breaking changes, but as soon as it is the internal module of the model I suppose it is fine to change its signature. 🚨 can be added to PR message if it is still required
> 
> Hi @ArthurZucker @qubvel,
> 
> Hope you're doing well, could you find some time today or tomorrow to provide guidance on this PR? Would love to check it off of my list!

Hi @ArthurZucker @qubvel any chance we can move forward with this PR this week? Happy to do whatever you would like, just need to get guidance on what you would like to do here.

**Comment by qubvel** [comment]
Hi @philkuz, sorry for the delay, the team was on the offsite this week. I will ping Arthur to get it reviewed and merged. Thanks for the patience

**Comment by philkuz** [comment]
> Hi @philkuz, sorry for the delay, the team was on the offsite this week. I will ping Arthur to get it reviewed and merged. Thanks for the patience

Thank you, Pavel!

File: src/transformers/models/mask2former/modeling_mask2former.py
Comment (ArthurZucker) [reply src/transformers/models/mask2former/modeling_mask2former.py:929]: Usually even if a module is `Internal` people still end up using it 😅 
This IS a breaking change, but acceptable IMO. Let's use  some  🚨 on the PR title to make sure we communicate about it on the release!

**Review by ArthurZucker — APPROVED** [review]
Thanks for this contribution! 🤗 Looks good will just update the PR name!

**Comment by qubvel** [comment]
cc @guangy10

**Comment by guangy10** [comment]
@philkuz Awesome! Thanks for expanding the export coverage to more models! 🚀 🚀 🚀

**Pull request merged**
